frontend/src/components/Menu/Menu.tsx

```diff
@@ -69,6 +68,18 @@ const LeftMenu: React.FC = () => {
         return labels.join('/');
     };
 
+    const handleLogin = () => {
+        const redirectUri = `${window.location.origin}/auth/callback`;
+        window.location.href = `${KEYCLOAK_LOGIN_URL}&redirect_uri=${encodeURIComponent(redirectUri)}`;
+    };
+
+    const handleLogout = () => {
+        const logoutUrl = `${KEYCLOAK_LOGOUT_URL}?redirect_uri=${encodeURIComponent(window.location.origin)}`;
+        localStorage.removeItem('access_token');
+        localStorage.removeItem('refresh_token');
+        window.location.href = logoutUrl;
```

Comment:
I’m not sure about this, but wouldn’t it be better to include login and logout in useAuth?

Reply:
https://www.youtube.com/watch?v=wLg04uu2j2o

Reply:
I'm not sure if this is exactly returning

Reply:
What it's called in this godforsaken language I don't know, but to my mind it's returning in pure form

Reply:
Done, but at what cost ... Because of this part of the code, I will never again look in the mirror as I did before

Reply:
![samuel-vs-js](https://github.com/user-attachments/assets/3b6f5291-a4eb-41af-a89e-9a49c8a4ef90)